config.yaml

```diff
@@ -54,6 +54,14 @@ options:
       to attempt auto-retrieval of intermediate certificates.  The default
       (false) is recommended for all production kubernetes installations, and
       any environment which does not have outbound Internet access.
+  ingress-default-ssl-certificate:
+    type: string
+    default: ""
+    description: |
+      Secret containing a SSL certificate to be used by the default HTTPS server
```

Comment:
The configuration should take SSL certificate as configuration input and templates/ingress-daemon-set.yaml need to be updated to create a secret if ingress-default-ssl-certificate is provided.

Reply:
What is it that you want me to change?

Just for clarification, if this is not set it will just default as usual to the self-signed internals certificate in the docker image.

That option takes only a secret/tls or empty.

Reply:
ingress-default-ssl-certificate should take a pem file as input parameter (or empty to default to fake certificate) and a secret need to be constructed in the charm here[1] and the secret can be passed to CMD arguments for /nginx-ingress-controller
Please see the discussion on launchpad bug [2]

[1] https://github.com/charmed-kubernetes/charm-kubernetes-worker/blob/master/templates/ingress-daemon-set.yaml
[2] https://bugs.launchpad.net/charm-kubernetes-worker/+bug/1859156